src/components/Experience.test.ts

```diff
@@ -0,0 +1,160 @@
+                `Wild ${team.selectedActor.nickname.join("")} appeared!`,
+        },
+    });
+
+    charmander.statistics.attack.current = 999;
+    skipBattle(clock, fsp, player);
+};
+
+const createGame = (charmanderLevel: number) => {
+    const { fsp, player, clock } = stubBlankGame({
+        automaticallyAdvanceMenus: true,
+    });
+
+    const charmander = fsp.equations.newPokemon({
+        level: charmanderLevel,
+        title: "CHARMANDER".split(""),
+        moves: [
+            {
+                remaining: 10,
+                title: "Scratch",
+                uses: 10,
+            },
+        ],
+    });
+
+    fsp.itemsHolder.setItem(fsp.storage.names.pokemonInParty, [
+        charmander,
+    ]);
+
+    const enemyPokemon: IPokemon = fsp.equations.newPokemon({
+        level: 1,
+        title: "PIDGEY".split(""),
+        moves: [
+            {
+                remaining: 10,
+                title: "Growl",
+                uses: 10,
+            },
+        ],
+    });
+
+    return{ fsp, player, clock, charmander, enemyPokemon };
+};
+describe("Experience", () => {
+    it("levels a Pokemon up through the levelup function", (): void => {
+        // Arrange
+        const { fsp } = stubBlankGame();
+        const pokemon: IPokemon = fsp.equations.newPokemon({
+            level: 5,
+            title: "CHARMANDER".split(""),
+        });
+
+        // Act
+        fsp.experience.levelup(pokemon);
+
+        // Assert
+        expect(pokemon.level).to.be.equal(6);
+    });
+
+    it("makes a Pokemon gain experience through the gainExperience function call", (): void => {
+        // Arrange
+        const { fsp } = stubBlankGame();
+        const pokemon: IPokemon = fsp.equations.newPokemon({
+            level: 5,
+            title: "CHARMANDER".split(""),
+        });
+
+        // Act
+        const result = fsp.experience.gainExperience(pokemon, 50);
+
+        // Assert
+        expect(pokemon.experience).to.be.equal(175);
+    });
+
+    it("doesn't level up a pokemon when it doesn't gain enough experience to do so", (): void => {
+        // Arrange
+        const { fsp } = stubBlankGame();
+        const pokemon: IPokemon = fsp.equations.newPokemon({
+            level: 5,
+            title: "CHARMANDER".split(""),
+        });
+
+        // Act
+        const result = fsp.experience.gainExperience(pokemon, 50);
+
+        // Assert
+        expect(result).to.be.equal(false);
+    });
+
+    it("levels up a pokemon when it gains enough experience to do so", (): void => {
+        // Arrange
+        const { fsp } = stubBlankGame();
+        const pokemon: IPokemon = fsp.equations.newPokemon({
+            level: 5,
+            title: "CHARMANDER".split(""),
+        });
+
+        // Act
+        const result = fsp.experience.gainExperience(pokemon, 100);
+
+        // Assert
+        expect(pokemon.level).to.be.equal(6);
+        expect(result).to.be.equal(true);
```

Comment:
This test and the one before still needs to be split up into two tests each: one that asserts the Pokemon's level and one that asserts the return value.

I know that seems trivial, but it's actually very important that unit tests each test a single (1) unit of work. People (myself included) often use unit tests as a sort of checklist for changing things: any failing unit test shows something you've broken. When a single unit test tests multiple things, it will fail when either fails, so it's less accurate as a checklist. 

Using this test as an example: if someone breaks whether the Pokemon's level changes, this test will fail and not give any indication on whether the return value is correct.